stratum/hal/lib/barefoot/bf_switch.cc

```diff
@@ -101,9 +105,30 @@ ::util::Status BFSwitch::VerifyChassisConfig(const ChassisConfig& config) {
   return status;
 }
 
+namespace {
+// Parses the P4 ForwardingPipelineConfig to check the format of the
+// p4_device_config. If it uses a newer Stratum format, this method converts
+// it to the legacy format used by the Barefoot PI implementatation. Otherwise,
+// the provided value is used.
+::util::Status ParseForwardingPipelineConfig(
```

Comment:
Rename to `CanonicalizeForwardingPipelineConfig`. Parsing implies an input and an output parameter. This is more of a transformation.